sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSquidSensor.java

```diff
@@ -79,9 +86,45 @@ public void execute(SensorContext context) {
     sonarComponents.setSensorContext(context);
     sonarComponents.registerCheckClasses(CheckList.REPOSITORY_KEY, CheckList.getJavaChecks());
     sonarComponents.registerTestCheckClasses(CheckList.REPOSITORY_KEY, CheckList.getJavaTestChecks());
+
+    ExecutorService executorService = createExecutor();
+    CompletionService cs =  new ExecutorCompletionService(executorService);
+    int fileSize = 0;
+    for (File sourceFile : getSourceFiles()) {
+      cs.submit(() -> scan(context, sourceFile, false), null);
+      fileSize++;
+    }
+    for (File testFile : getTestFiles()) {
+      cs.submit(() -> scan(context, testFile, true), null);
+      fileSize++;
+    }
+    executorService.shutdown();
+    for (int i = 0; i < fileSize; i++) {
+      try {
+        cs.take().get();
+      } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142)